### ISSUE TITLE
OpenApiPaginator

### ISSUE THREAD

**Issue by MasaKni** [cnizzardini/cakephp-swagger-bake#562]
I was thinking about making the OpenApiPaginator accessible to be used on other methods that use pagination,
also adding a possibility to extend it and adding custom parameters, for instance i would like to add an order parameter that sorts by an array because currently it only allows sorting by one field. 

so at least we would need the definePagination function to be protected 

what do you think? 

```
 protected function definePagination(): void
  {
      parent::definePagination();
      $this->pushOrderParameter();
  }

  private function pushOrderParameter()
  {
  }
```

**Comment by cnizzardini** [comment]
Yeah I think making paginator available on non-index methods is a good idea. When I originally developed this, given the amount of code necessary, only having it work on index() methods was a compromise. Unfortunately I haven't ever gotten around to it.

**Comment by MasaKni** [comment]
#563

**Comment by cnizzardini** [comment]
@MasaKni thank you, your changes have been released: https://github.com/cnizzardini/cakephp-swagger-bake/releases/tag/v3.1.0
